specification/azurefleet/AzureFleet.Management/baseVirtualMachineProfile.tsp

```diff
@@ -0,0 +1,2515 @@
+   * Minimum api-version: 2022-11-01
+   */
+  serviceArtifactReference?: ServiceArtifactReference;
+
+  /**
+   * Specifies the security posture to be used for all virtual machines in the scale
+   * set. Minimum api-version: 2023-03-01
+   */
+  securityPostureReference?: SecurityPostureReference;
+
+  /**
+   * Specifies the time in which this VM profile for the Virtual Machine Scale Set
+   * was created. Minimum API version for this property is 2023-09-01. This value
+   * will be added to VMSS Flex VM tags when creating/updating the VMSS VM Profile
+   * with minimum api-version 2023-09-01.
+   */
+  @visibility("read")
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  timeCreated?: utcDateTime;
+}
+
+/**
+ * Describes a virtual machine scale set OS profile.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" ""
+model VirtualMachineScaleSetOSProfile {
+  /**
+   * Specifies the computer name prefix for all of the virtual machines in the scale
+   * set. Computer name prefixes must be 1 to 15 characters long.
+   */
+  computerNamePrefix?: string;
+
+  /**
+   * Specifies the name of the administrator account. <br><br> **Windows-only
+   * restriction:** Cannot end in "." <br><br> **Disallowed values:**
+   * "administrator", "admin", "user", "user1", "test", "user2", "test1", "user3",
+   * "admin1", "1", "123", "a", "actuser", "adm", "admin2", "aspnet", "backup",
+   * "console", "david", "guest", "john", "owner", "root", "server", "sql",
+   * "support", "support_388945a0", "sys", "test2", "test3", "user4", "user5".
+   * <br><br> **Minimum-length (Linux):** 1  character <br><br> **Max-length
+   * (Linux):** 64 characters <br><br> **Max-length (Windows):** 20 characters
+   */
+  adminUsername?: string;
+
+  @doc("""
+    Specifies the password of the administrator account. <br><br> **Minimum-length
+    (Windows):** 8 characters <br><br> **Minimum-length (Linux):** 6 characters
+    <br><br> **Max-length (Windows):** 123 characters <br><br> **Max-length
+    (Linux):** 72 characters <br><br> **Complexity requirements:** 3 out of 4
+    conditions below need to be fulfilled <br> Has lower characters <br>Has upper
+    characters <br> Has a digit <br> Has a special character (Regex match [\\W_])
+    <br><br> **Disallowed values:** \"abc@123\", \"P@$$w0rd\", \"P@ssw0rd\",
+    \"P@ssword123\", \"Pa$$word\", \"pass@word1\", \"Password!\", \"Password1\",
+    \"Password22\", \"iloveyou!\" <br><br> For resetting the password, see [How to
+    reset the Remote Desktop service or its login password in a Windows
+    VM](https://docs.microsoft.com/troubleshoot/azure/virtual-machines/reset-rdp)
+    <br><br> For resetting root password, see [Manage users, SSH, and check or
+    repair disks on Azure Linux VMs using the VMAccess
+    Extension](https://docs.microsoft.com/troubleshoot/azure/virtual-machines/troubleshoot-ssh-connection)
+    """)
+  adminPassword?: string;
```

Comment:
They are `x-ms-secret` in Swagger https://github.com/Azure/azure-rest-api-specs/blob/main/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/computeRPCommon.json#L1502

Should it be `@visibility("create", "update")`?

Reply:
Good point. Addressed this.